plugins/catalog-backend/src/service/DefaultEntitiesCatalog.ts

```diff
@@ -97,7 +97,7 @@ function addCondition(
   // make a lot of sense. However, it had abysmal performance on sqlite
   // when datasets grew large, so we're using IN instead.
   const matchQuery = db<DbSearchRow>('search')
-    .select(entityIdField)
+    .select('search.entity_id')
```

Comment:
we were wondering why the prepending of search was needed?

Reply:
Ah good call, I should have written a comment about that in the PR! This is indeed not clear why it's here.

I think the old code was essentially flat out wrong. This select wants to find a set of entity IDs, _specifically from the search table_, and then say basically `... WHERE <entityIdField> IN <that set of IDs from the search table>`.

The old code happened to work basically by accident, due to how the join worked, because it actually grabbed the entity IDs from the "other" thing that was being filtered in the first place. But now in my PR, the nesting in the outer query became a bit more complex, such that the SQL engine started saying that the column name `entity_id` was ambiguous since it occurs in several tables that are part of the overall larger query. So I made this precisely target the search table instead, which was the original intent as far as I can tell.